packages/app/src/composables/useWallet.ts

```diff
@@ -0,0 +1,269 @@
+import { type ComputedRef, reactive, toRefs, type ToRefs, unref } from "vue";
+
+import detectEthereumProvider from "@metamask/detect-provider";
+import { type RemovableRef, useStorage } from "@vueuse/core";
+import { ethers } from "ethers";
+import * as zkSyncSdk from "zksync-ethers";
+
+import defaultLogger from "./../utils/logger";
+
+import type { BaseProvider } from "@metamask/providers";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type JsonRpcError = any;
+
+type WalletState = {
+  isReady: boolean;
+  isConnectPending: boolean;
+  isConnectFailed: boolean;
+  isMetamaskInstalled: boolean;
+  address: string | null;
+};
+
+type UseWallet = ToRefs<WalletState> & {
+  initialize: () => Promise<void>;
+
+  connect: () => Promise<void>;
+  disconnect: () => void;
+
+  getL1Signer: () => Promise<zkSyncSdk.L1Signer>;
+  getL2Signer: () => Promise<zkSyncSdk.Signer>;
+};
+
+export type NetworkConfiguration = {
+  l1ChainId: number;
+  l2ChainId: number;
+  rpcUrl: string;
+  explorerUrl: string;
+  chainName: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function processException(e: any, message: string): never {
+  if (e instanceof WalletError) {
+    throw e;
+  } else if (e?.code === 4001 || e?.code === "ACTION_REJECTED") {
+    throw WalletError.TransactionRejected();
+  } else if (e?.code === -32603) {
+    throw WalletError.JsonRpcError(e?.data?.data?.message?.length ? e.data.data.message : e.message);
+  } else if (e?.code === "SERVER_ERROR") {
+    throw WalletError.InternalServerError();
+  }
+  throw WalletError.UnknownError(e?.message?.length ? e.message : message);
+}
+
+const state = reactive<WalletState>({
+  isReady: false,
+  isMetamaskInstalled: false,
+  isConnectPending: false,
+  isConnectFailed: false,
+
+  address: null,
+});
+
+export const isAuthenticated: RemovableRef<boolean> = useStorage<boolean>("useWallet_isAuthenticated", false);
+
+export default (
```

Comment:
I think it's a good practice to not export it as default but rather named export, so that it automatically shows up in vscode when you start typing the name

Reply:
All existing composables in the project are exported as a default. I would rather keep this one in the same style, to keep things consistent.